dashboard/app/serializers/api/v1/pd/application_serializer.rb

```diff
@@ -93,7 +94,66 @@ def attending_teachercon
     object&.workshop&.teachercon?
   end
 
-  def principal_approval
+  def principal_approval_state
     object.try(:principal_approval)
   end
+
+  def school_stats
+    data = nil
+    if object.try(:school_id)
```

Comment:
Suggest writing this as

`return nil unless object.try(:school_id)`
followed by
`return nil unless stats`

Less indenting, easier to read